src/realm/db.cpp

```diff
@@ -2412,10 +2412,15 @@ Replication::version_type DB::do_commit(Transaction& transaction, bool commit_to
     version_type new_version = current_version + 1;
 
     if (!transaction.m_objects_to_delete.empty()) {
+        std::map<TableKey, std::vector<ObjKey>> table_object_map;
         for (auto it : transaction.m_objects_to_delete) {
-            transaction.get_table(it.table_key)->remove_object(it.obj_key);
+            // Checking ttl would go here
```

Comment:
this would be lazy checking

Reply:
But I guess if we ever get this implemented, this would be the place to check which objects were due to be deleted. BTW if we skip recording the time the object was created, we can make this even more efficient. @danieltabacaru should we do that?

Reply:
I am not saying it's a bad place to check, it's just that we would be relying on other things being committed. I am not if it's the same idea, but we don't really need the object keys, the table keys is enough (we can then clear the tables). we can optimize it further by also skipping storing the table keys by iterating through the tables and clearing the asymmetric ones.

Reply:
Of course - the table keys are sufficient. Even better!